utils/rpms/packaging/get_base_branch

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# find the base branch of the current branch
+
+set -eux -o pipefail
+
+ORIGIN=origin
+mapfile -t all_bases < <(echo "master"; git branch -r |
+  sed -ne "/^  $ORIGIN\\/release\\/[0-9]/s/^  $ORIGIN\\///p")
+TARGET="master"
+min_diff=-1
+for base in "${all_bases[@]}"; do
+    git rev-parse --verify "$ORIGIN/$base" &> /dev/null || continue
+    commits_ahead=$(git log --oneline "$ORIGIN/$base..HEAD" | wc -l)
+    if [ "$min_diff" -eq -1 ] || [ "$min_diff" -gt "$commits_ahead" ]; then
+        TARGET="$base"
+        min_diff=$commits_ahead
+    fi
```

Comment:
As we discussed, checking `commits_behind` like `find_base.sh` does is only for edge cases (if even useful at all). Could we either (A) update `find_base.sh` to be in-sync or (B) share the function?

Also, is `utils/rpms/packaging` the right place for this, if only called from the Jenkinsfile?

Reply:
> As we discussed, checking `commits_behind` like `find_base.sh` does is only for edge cases (if even useful at all).

Right, which is why I omitted that in this implementation.

> Could we either (A) update `find_base.sh` to be in-sync or (B) share the function?

My original intent was to share/genericize the function but you seemed to want to keep the `commits_behind` functionality in `find_base.sh` and I didn't really see the need for it here.  Indeed, it nags at me that it might make finding the true base (where we disregard intermediate branches such as `features/*`) actually less accurate.

> Also, is `utils/rpms/packaging` the right place for this, if only called from the Jenkinsfile?

It seems odd, yes, but this functionality is used across projects (`daos` and all of the dependency packaging projects as it's used in the dependency packaging `pipeline`s) and `packaging/` is the common denominator there -- without inventing yet another cross-project library.

Reply:
FWIW, I'd rather give up `commits_behind` in `find_base.sh` than have two slightly different versions.

Reply:
> this functionality is used across projects

It is currently, or it will be after landed?

Reply:
There are a few moving parts that are working towards this, but it will be used by all project pipelines that build RPMs (hence the association with `packaging/`.

Reply:
I do think it's better for this function to be in `daos` instead of `pipeline-lib` - for re-use purposes as well as ease-of-updating.

The question of "where do I put a script that is used for CI, hooks, etc." has come up before, and I'm not sure where the best place is to put them. I don't want to block on this, so I'm okay with what you have and we can revisit later if/when we start re-using this function elsewhere